psh/tools/common.py

```diff
@@ -38,6 +41,30 @@ def create_testdir(p, dirname):
     psh.assert_cmd(p, f'mkdir {dirname}', '', msg)
 
 
+def assert_mtime_change(p, files_to_check, datetimes: List[datetime], dir=''):
+    ''' Asserts that mentioned files have modification time equal to passed datetime with 1 min margin.
+    Year and seconds are not checked.
+    The `files_to_check` arg has to be a list of file names from the `dir` directory.'''
+    existing_files = psh.ls(p, dir)
+    existing_fnames = [file.name for file in existing_files]
+    assert set(files_to_check).issubset(set(existing_fnames)), 'Files to check are not present in the passed directory!'
+
+    for fname in files_to_check:
+        target_datetime = datetimes[fname]
+        existing_file_id = existing_fnames.index(fname)
+        file_datetime = existing_files[existing_file_id].datetime
+
+        # to prevent failed assertion when typing date in 00:59 and touch in 01:00
+        if file_datetime.minute != target_datetime.minute and abs(file_datetime.minute - target_datetime.minute) == 1:
```

Comment:
To cover 59:59 and 00:00 case, the `date.minute` is then lower than `target_datetime.minute`

Reply:
Yeap you are right, but the date.minute is int type, so the difference will be 59. You might use `-` operator on datetime type and  then check particular fields.

Reply:
Yeah, I've finally done it this way:

```python
        # to prevent failed assertion when typing date in 00:59 and touch in 01:00
        if date - target_datetime == timedelta(minutes=1):
            date = target_datetime
```